articles/azure-maps/elevation-data-services.md

```diff
@@ -0,0 +1,380 @@
+    from wsgiref import headers
+    import azure.functions as func
+    import psycopg2
+    # Database to connect to
+    DATABASE = {
+        'user':     'postgres',
+        'password': '{password}',
+        'host':     'localhost',
+        'port':     '5432',
+        'database': '{database}'
+        }
+    def main(req: func.HttpRequest) -> func.HttpResponse:
+        logging.info('Python HTTP trigger function processed a request.')
+        DATABASE_CONNECTION = None
+        # get url parameters http://localhost:7071/api/tileserver?zoom={z}&x={x}&y={y}
+        # http://localhost:7071/api/tileserver?zoom=16&x=10556&y=22870
+        zoom = int(req.params.get('zoom'))
+        x = int(req.params.get('x'))
+        y = int(req.params.get('y'))
+        table = req.params.get('table')
+        # calculate the envelope of the tile
+        # Width of world in EPSG:3857
+        worldMercMax = 20037508.3427892
+        worldMercMin = -1 * worldMercMax
+        worldMercSize = worldMercMax - worldMercMin
+        # Width in tiles
+        worldTileSize = 2 ** zoom
+        
+        # Tile width in EPSG:3857
+        tileMercSize = worldMercSize / worldTileSize
+        
+        # Calculate geographic bounds from tile coordinates
+        # XYZ tile coordinates are in "image space" so origin is
+        # top-left, not bottom right
+        xmin = worldMercMin + tileMercSize * x
+        xmax = worldMercMin + tileMercSize * (x + 1)
+        ymin = worldMercMax - tileMercSize * (y + 1)
+        ymax = worldMercMax - tileMercSize * y
+        # Generate SQL to materialize a query envelope in EPSG:3857.
+        # Densify the edges a little so the envelope can be
+        # safely converted to other coordinate systems.
+        DENSIFY_FACTOR = 4
+        segSize = (xmax - xmin)/DENSIFY_FACTOR
+        sql01 = 'ST_Segmentize(ST_MakeEnvelope(' + str(xmin) + ', ' + str(ymin) + ', ' + str(xmax) + ', ' + str(ymax) + ', 3857), ' + str(segSize) +')'
+       
+        # Generate a SQL query to pull a tile worth of MVT data
+        # from the table of interest. 
+        # Materialize the bounds
+        # Select the relevant geometry and clip to MVT bounds
+        # Convert to MVT format
+        # if table == 'kc_parcels':
+        #     sql02 = 'WITH bounds AS (SELECT ' + sql01 + ' AS geom, ' + sql01 + '::box2d AS b2d), mvtgeom AS (SELECT ST_AsMVTGeom(ST_Transform(t.geom, 3857), bounds.b2d) AS geom, addr_full FROM kc_parcels t, bounds WHERE ST_Intersects(t.geom, ST_Transform(bounds.geom, 4326))) SELECT ST_AsMVT(mvtgeom.*) FROM mvtgeom' 
+        # elif table == 'kc_school_districts':
+        #     sql02 = 'WITH bounds AS (SELECT ' + sql01 + ' AS geom, ' + sql01 + '::box2d AS b2d), mvtgeom AS (SELECT ST_AsMVTGeom(ST_Transform(t.geom, 3857), bounds.b2d) AS geom FROM kc_school_districts t, bounds WHERE ST_Intersects(t.geom, ST_Transform(bounds.geom, 4326))) SELECT ST_AsMVT(mvtgeom.*) FROM mvtgeom' 
+        # elif table == 'contourlines_smooth':
```

Comment:
Please remove the following commented code

# if table == 'kc_parcels':
        #     sql02 = 'WITH bounds AS (SELECT ' + sql01 + ' AS geom, ' + sql01 + '::box2d AS b2d), mvtgeom AS (SELECT ST_AsMVTGeom(ST_Transform(t.geom, 3857), bounds.b2d) AS geom, addr_full FROM kc_parcels t, bounds WHERE ST_Intersects(t.geom, ST_Transform(bounds.geom, 4326))) SELECT ST_AsMVT(mvtgeom.*) FROM mvtgeom' 
        # elif table == 'kc_school_districts':
        #     sql02 = 'WITH bounds AS (SELECT ' + sql01 + ' AS geom, ' + sql01 + '::box2d AS b2d), mvtgeom AS (SELECT ST_AsMVTGeom(ST_Transform(t.geom, 3857), bounds.b2d) AS geom FROM kc_school_districts t, bounds WHERE ST_Intersects(t.geom, ST_Transform(bounds.geom, 4326))) SELECT ST_AsMVT(mvtgeom.*) FROM mvtgeom' 
        # elif table == 'contourlines_smooth':